pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -220,15 +226,10 @@ func (pl *DefaultPreemption) FindCandidates(ctx context.Context, state *framewor
 		klog.Infof("from a pool of %d nodes (offset: %d, sample %d nodes: %v), ~%d candidates will be chosen", len(potentialNodes), offset, len(sample), sample, numCandidates)
 	}
 	candidates, nodeStatuses := dryRunPreemption(ctx, pl.fh, state, pod, potentialNodes, pdbs, offset, numCandidates)
-	// Return a FitError only when there are no candidates that fit the pod.
-	if len(candidates) == 0 {
-		return candidates, &framework.FitError{
-			Pod:                   pod,
-			NumAllNodes:           len(potentialNodes),
-			FilteredNodesStatuses: nodeStatuses,
-		}
+	for node, status := range unschedulableNodeStatus {
+		nodeStatuses[node] = status
```

Comment:
what happens if a node is a part of both maps? should we overwrite the dryRunPreemption status with the unschedulableNodeStatus value?

Reply:
Not sure if I understand your question correctly, the `unschedulableNodeStatus` we got here are from the nodes that are have not evaluated by `dryRunPreemption`. 

`len(unschedulableNodeStatus) + len(potentialNodes) = allNodes`

I cannnot see where the overwrite could happen.